### DISCUSSION TITLE
Support for importing a build.log

### DISCUSSION THREAD

**Discussion by ryzngard** [stryker-mutator/stryker-net#3143] [Q&A]
I was evaluating if stryker-net could be run on https://github.com/dotnet/razor and with my very limited knowledge it just doesn't. Trying to debug I'm guessing it's because of the heavy use of variable replacement in project files for things like target framework. I see a lot of code related to parsing project/solution files and was wondering: would it be easier to support a structure log output from something like `dotnet build -bl` and read in the necessary properties. That would give the complete requirements for what final evaluations are when restore/compile happens.

**Comment by rouke-broersma** [comment]
Hi @ryzngard 

We primarily use buildalyzer for project information discovery. Buildalyzer executes msbuild against the project file and reads the output, so variable replacement is not an issue. Buildalyzer actually supports msbuildlog as well, but I don't think this is more useful than running Buildalyzer directly.

I'm not sure if buildalyzer has support for razor projects, I would look there first.

**Reply by ryzngard** [reply]
The tooling for razor is just a normal dotnet project that builds into either a cmd project as an lsp server or gets packaged as a vsix.

Is buildalyzer a separate tooling system? I can take a look there

**Reply by ryzngard** [reply]
Specifically I think the msbuild command to build might have been wrong. I'll dig into that though and see. Something definitely was missing and throwing after buildalyzer ran

**Reply by rouke-broersma** [reply]
Buildalyzer can be found here: https://github.com/phmonte/Buildalyzer
